frontend/ui/widget/spinwidget.lua

```diff
@@ -33,6 +33,9 @@ local SpinWidget = InputContainer:new{
     value_hold_step = 4,
     ok_text = _("OK"),
     cancel_text = _("Cancel"),
+    -- extra button on bottom
+    custom_text = nil,
+    callback_custom = nil,
```

Comment:
nitpick, but either `custom_` as a prefix for both, or as a suffix for both :) it looks strange as they are there.
(and may be `extra` instead of `custom` as you name that button "extra button" - unless this kind of added stuff is already called custom elsewhere.)

Reply:
Prefix is probably more in line with the rest of the code around the program, good catch.